pressbooks/pressbooks-book-directory-fe
# Read more action for book's descriptions

Related issue: https://github.com/pressbooks/pressbooks-book-directory-fe/issues/285

We were using CSS to display till 6 lines in the book cards for long descriptions. To the vue component level, we just display the regular `item.description` Algolia's response, and then the CSS trick did the rest.
The requirement specifies that we must show 6 lines, which means we can not use number of characters to decide directly the string length to be displayed, since it changes according to the client screen resolution.

I added the [vue-clamp](https://justineo.github.io/vue-clamp) library, which does the job, and provides slots to add the toggle action to show the complete description.

I also added a new test in `bookCards.spec.js`. (Note the temporary `wait` until will solve the cypress sync issue).

A made some changes in other components to fix linting issues.

## Test
- Run `npm install` to install the new library
- Descriptions by default must display 6 lines, if present in the book card.
- For longer descriptions, it should be truncated with the `...` appended. Also a new `Read more` link must be displayed.
- Clicking on `Read more` link, the book cards must be auto expaneded and display the complete book's description.

- Run E2E the `bookCards.scpec.js` test.
